tests/karate/docs/README.md

```diff
@@ -0,0 +1,172 @@
+
+## Architecture
+### Features and Scenarios
+Karate test cases are called scenarios and they are grouped within different feature files. Each feature file tests a different message type (i.e. electionOpen, createRollCall etc.). 
+
+Scenarios are written in the Gerkhin syntax using the following keywords:
+
+- **Given**: Prepare the JSON payload to be sent to the component being tested. 
+- **When**: Defines the action that is to be performed with the payload. For instance, `publish` creates a message of type publish that contains some high-level message data, or `send` to send raw JSON data. 
+- **And**: Connector that can be used after any of the other keywords.
+- **Then**: Asserts that the action taken in the 'When' step has the expected outcome.
+
+### Background section
+Code defined in the background section of a feature file runs before each scenario. This is especially useful for: 
+
+- **Sharing scopes with other features**: The call to `read(classpath: "path/to/feature")` is used to make the current feature share the same scope as the feature that is called. This means they share definitions (def variables) and configurations. JavaScript functions defined in the called feature are also available for use in the current feature's context. For instance, reading `mockClient.feature` exposes functions like `createMockFrontend` and `createMockBackend` that can be used to create mock components.
+
+- **Setting up previous steps necessary for a test**: For instance, before roll call messages can be tested, a LAO needs to be created first. `simpleScenarios.feature` contains many such useful setup steps.
+
+### Data model
+To generate valid message data for JSON payloads dynamically, a simplified version of the model is implemented in Java code. Mock components can create valid objects (for instance LAO, RollCall, Elections etc.), that can be used to handcraft messages. These objects also provide functions to override their own data to some invalid values, to craft invalid messages. 
+
+Some care needs to be taken if more of these functions to override valid data are implemented in the future. For example, when setting an invalid LAO name to test if the server rejects this, the LAO id also needs to be recomputed. The LAO id depends on the LAO name, and if the id is not recomputed the test might fail due to invalid LAO id, and not because the name was invalid. Currently, there is no way to distinguish this as only the error code is asserted and not the error message. This is a possible improvement that could be done in the future. 
```

Comment:
> Currently, there is no way to distinguish this as only the error code is asserted and not the error message

You can "future-proof" this by saying "As of February 2024, ...", which will make it obvious, later on, if this documentation needs to be taken with a pinch of salt or updated.